Source/Scene/CesiumIon.js

```diff
@@ -130,16 +173,36 @@ define([
      *   });
      */
     CesiumIon.createImageryProvider = function(assetId, options) {
-        return CesiumIon.createResource(assetId, options)
-            .then(function(resource) {
-                return resource.createImageryProvider();
+        var endpointResource = CesiumIon._createEndpointResource(assetId, options);
+
+        return CesiumIon._loadJson(endpointResource)
+            .then(function(endpoint) {
+
+                if (endpoint.type !== 'IMAGERY') {
+                    throw new RuntimeError('Cesium ion asset ' + assetId + ' is not an imagery asset.');
+                }
+
+                var externalType = endpoint.externalType;
+                if (!defined(externalType)) {
+                    return createTileMapServiceImageryProvider({
+                        url: CesiumIonResource.create(endpoint, endpointResource)
+                    });
+                }
+
+                var factory = ImageryProviderMapping[externalType];
+
+                if (!defined(factory)) {
+                    throw new RuntimeError('Unrecognized Cesium ion imagery type: ' + externalType);
+                }
+
+                return factory(endpoint.options);
             });
     };
 
     /**
      * A {@link Resource} instance that encapsulates Cesium ion asset
-     * creation and automatic refresh token handling.  This object
-     * should not be created directly, use CesiumIonResource.create
+     * creation and automatic refresh token handling. This object
+     * should not be created directly, use CesiumIonResource.create.
      *
      * @private
      */
```

Comment:
Should `CesiumIonResource` be in its own file? Usually in Cesium if we have a class with a few functions, we usually split it out.

Reply:
I moved it, my main reason for keeping it off is because it ends up on the `Cesium` object in the combined version, but that's par for the course for us anyway.  This should be good to go.